packages/example-application/src/form.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useCallback } from 'react';
+import * as GovUK from 'govuk-react';
+import { Link } from 'react-router-dom';
+
+export default () => {
+  const [firstName, setFirstName] = useState();
+  const [errors, setErrors] = useState();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const handleSubmit = useCallback(() => {
+    if (isSubmitting) return;
+    if (!firstName) {
+      setErrors({ firstName: 'Please enter a first name' });
+    } else {
+      setIsSubmitting(true);
+      setTimeout(() => {
+        setIsSubmitted(true);
+        setIsSubmitting(false);
+      }, 1000);
+    }
+  });
```

Comment:
hmm - eslint should have been whining at you about the missing dependencies list for this function :grin:

Reply:
https://github.com/govuk-react/govuk-react/issues/827